tools/self-hosted-runner-utils/cloudwatch_logs.py

```diff
@@ -0,0 +1,327 @@
+    log_group_name = get_log_group_name(options.function_name, options.log_group)
+
+    # Parse time arguments if provided
+    start_time = None
+    end_time = None
+    if options.start_time:
+        start_time = parse_timestamp(options.start_time)
+    if options.end_time:
+        end_time = parse_timestamp(options.end_time)
+
+    if options.dry_run:
+        print(f"DRY RUN: Would download logs from {log_group_name}")
+        if options.output_file:
+            print(f"Output would be written to: {options.output_file}")
+        return
+
+    try:
+        # Initialize CloudWatch Logs client
+        logs_client = boto3.client("logs")
+
+        if options.tail:
+            stream_logs(logs_client, log_group_name, start_time, options.poll_interval)
+            return
+
+        # Get latest log streams
+        log_streams = get_latest_log_streams(logs_client, log_group_name)
+
+        if not log_streams:
+            print(f"No log streams found in {log_group_name}")
+            return
+
+        all_logs = []
+
+        # Download logs from each stream
+        for stream in tqdm(log_streams, desc="Downloading logs"):
+            stream_name = stream["logStreamName"]
+
+            events, _ = download_log_events(
+                logs_client,
+                log_group_name,
+                stream_name,
+                start_time,
+                end_time,
+                options.lines // len(log_streams)
+                if len(log_streams) > 1
+                else options.lines,
+            )
+
+            if events:
+                formatted_logs = format_log_events(events, stream_name)
+                all_logs.append(formatted_logs)
+
+        if not all_logs:
+            print("No log events found matching the criteria")
+            return
+
+        # Prepare header
+        header = [
+            f"CloudWatch Logs for {options.function_name}\n",
+            f"Log Group: {log_group_name}\n",
+            f"Downloaded: {datetime.now(timezone.utc).strftime('%Y-%m-%d %H:%M:%S UTC')}\n",
+            "=" * 80 + "\n\n",
+        ]
+
+        if options.output_file:
+            # Write to output file
+            with open(options.output_file, "w", encoding="utf-8") as f:
+                f.writelines(header)
+                for log_content in all_logs:
+                    f.write(log_content)
+            print(f"Downloaded logs to: {options.output_file}")
+        else:
+            # Write to stdout
+            import sys
+
+            sys.stdout.writelines(header)
+            for log_content in all_logs:
+                sys.stdout.write(log_content)
+
+    except NoCredentialsError:
+        print(
+            "Error: AWS credentials not found. Please configure your AWS credentials."
+        )
+    except ValueError as e:
+        print(f"Error: {e}")
+    except ClientError as e:
+        print(f"AWS Error: {e}")
+    except Exception as e:
+        print(f"Unexpected error: {e}")
```

Comment:
Again, what's the value of catching those errors? If script fails, it should fail and return non-zero status code, but right now it's not the case